Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -427,13 +427,17 @@ void FieldProbe::ComputeDiags (int step)
             // reset m_data vector to clear pushed values. Reserves data
             m_data.clear();
             m_data.shrink_to_fit();
-            m_data.reserve(numparticles * noutputs);
+            // the +1 below is to temporarily store the particle ID in order
+            // to sort the output
+            m_data.reserve(numparticles * (noutputs + 1));
```

Comment:
why all the +1 instead of removing the id in the final print to file? :)

Reply:
No strong reason. I just kept `noutputs` as its original meaning of how many quantities of interest are saved. I'll change it to have the +1 included in `noutputs`.